partner_contact_in_several_companies/models/res_partner.py

```diff
@@ -78,32 +77,33 @@ def create(self, vals):
         that the name on an attached contact is the same as the name on the
         contact it is attached to."""
         modified_self = self._basecontact_check_context("create")
-        if not vals.get("name") and vals.get("contact_id"):
-            vals["name"] = modified_self.browse(vals["contact_id"]).name
+        contact_id = vals.get("contact_id")
+        if not vals.get("name") and contact_id:
+            vals.update(name=modified_self.browse(contact_id).name)
         return super(ResPartner, modified_self).create(vals)
 
     def read(self, fields=None, load="_classic_read"):
-        modified_self = self._basecontact_check_context("read")
-        return super(ResPartner, modified_self).read(fields=fields, load=load)
+        return super(ResPartner, self._basecontact_check_context("read")).read(
+            fields=fields, load=load
+        )
 
     def write(self, vals):
-        modified_self = self._basecontact_check_context("write")
-        return super(ResPartner, modified_self).write(vals)
+        return super(ResPartner, self._basecontact_check_context("write")).write(vals)
 
     def unlink(self):
-        modified_self = self._basecontact_check_context("unlink")
-        return super(ResPartner, modified_self).unlink()
+        return super(ResPartner, self._basecontact_check_context("unlink")).unlink()
 
     def _compute_commercial_partner(self):
         """Returns the partner that is considered the commercial
         entity of this partner. The commercial entity holds the master data
         for all commercial fields (see :py:meth:`~_commercial_fields`)"""
-        result = super(ResPartner, self)._compute_commercial_partner()
+        super(ResPartner, self)._compute_commercial_partner()
         for partner in self:
             if partner.contact_type == "attached" and not partner.parent_id:
                 partner.commercial_partner_id = partner.contact_id
-        return result
+        return
```

Comment:
You could remove `return` because it's optional in this situation.
By default function returned None even return not set.